### DISCUSSION TITLE
GPSD client on windows, server on linux?

### DISCUSSION THREAD

**Discussion by Linwood-F** [Stellarium/stellarium#2172] [Q&A]
From trying to get it to work I suspect it just will not, but to confirm.... 

I have a 1PPS GPS server running on a rPi, and it runs gpsd (it's also my portable weather station so goes with me to a dark site). 

I use Stellarium on windows (with NINA). 

I can't take GPS from the rPi via gpsd from the windows program, right?   Not trying to put the GPS there, just read it?

**Comment by alex-w** [comment]
Define `gui/gpsd_hostname` in config.ini file to read gps from rpi host. Details in Stellarium User Guide.

**Comment by Linwood-F** [comment]
I did that, but it just said: 

`No connected devices found. NMEA GPS lookup failed.
`

I thought that meant despite gpsd_hostname and port being defined it was still looking (only) for the com port.

**Reply by alex-w** [reply]
Damn, you are right - on Windows Stellarium support NMEA GPS only. In the theory you may compile Stellarium with GPSD support on Windows, but it requires binary package of libgpsd.

**Reply by Linwood-F** [reply]
Ah, OK, will quit trying.  I actually found I wasn't running gpsd right so the port was open for ipv6 but not ipv4 (who knows why it defaulted like that).  Fixing that anyway, but I guess it's moot. 

Thank you.

**Reply by Linwood-F** [reply]
Though that said I should add, for closure, once my listen socket was right it still wouldn't work from windows, so that seems confirmed by trial as well.

**Reply by gzotti** [reply]
At least when I developed this, there was no useful GPSD port for Windows. But you can set up your RPi as time server in your network, i.e. sync your operating system with it. I cannot say where to tweak e.g. hourly corrections, though. And probably you can get some Bluetooth connection to transmit your GPS location data. If successful, we can put your instructions into the user guide.

**Reply by Linwood-F** [reply]
Yeah, that's already done, it was just a nice to have way at a dark site to pick up the lat/long in stellarium.  I already have it programmed in the sites in APCC which also flows to NINA.  Thank you.

**Reply by gzotti** [reply]
It would be an interesting task to try running gpsd on WSL, then Stellarium self-built on WSL might work, but you may lose hardware accelerated graphics. The connection to the Windows version of Stellarium would still be gpsd-less.
